core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -613,7 +613,7 @@ private void readCurrentBlockToPos(long pos) throws IOException {
     // Do not set the buffer size too small to avoid slowing down seek by too much.
     byte[] buffer = new byte[Math.min((int) mSeekBufferSizeBytes, (int) len)];
     do {
-      int bytesRead = read(buffer);
+      int bytesRead = read(buffer, 0, (int) Math.min(buffer.length, len));
```

Comment:
@peisun1115 if that's the case, please replace the whole code block from lines 612 to 619 with:

```
// Read the remainder of the block in mSeekBufferSizeBytes increments
byte[] buffer = new byte[mSeekBufferSizeBytes];
do {
  // account for the last read which might be less than mSeekBufferSizeBytes bytes
  int bytesRead = read(buffer, 0, (int) Math.min(buffer.length, len);
  Preconditions.checkState(bytesRead > 0, PreconditionMessage.ERR_UNEXPECTED_EOF);
  len -= bytesRead;
} while (len > 0);
```